bindings/cs/rl.net.native/rl.net.api_status.cc

```diff
@@ -18,4 +18,14 @@ API const char* GetApiStatusErrorMessage(reinforcement_learning::api_status* sta
 API int GetApiStatusErrorCode(reinforcement_learning::api_status* status)
 {
     return status->get_error_code();
-}
\ No newline at end of file
+}
+
+API void UpdateApiStatusSafe(reinforcement_learning::api_status* status, int error_code, const char* message)
+{
+    reinforcement_learning::api_status::try_update(status, error_code, message);
```

Comment:
I do not believe so: `api_status` takes a copy of `message` and holds on to it, as it is assumed that the caller could free the string coming into `try_update` whenever:

    static void try_update(api_status* status, int new_code, const char* new_msg);

The `api_status` object then owns freeing that copy - which it will do when it gets deleted. The incoming copy is the responsibility of the caller of `UpdateApiStatusSafe`.